env0/templates.go

```diff
@@ -99,11 +99,6 @@ func getTemplateSchema(templateType TemplateType) map[string]*schema.Schema {
 			Description: "terraform / terragrunt file folder inside source code",
 			Optional:    true,
 		},
-		"revision": {
```

Comment:
moved below.
Revision is only needed for the shared use-case.

Reply:
in the "single" use-case. use the environment revision.

Reply:
If it is needed shouldn't this be Optional: false? (probably just something I am not understanding properly)

Reply:
It's still Optional.
The point of this change is to use the Optional variable in the environment resource and not in the template resource. If I have left it as is there would be two revision variables (for the templateless use-case). 
(One revision in environment resource and one revision in template resource).

Reviewing it... I should probably remove the "computed" field in environment I'll take care that.

Reply:
@ItamarMalka -
I've reverted this change and placed back revision in the template resource (for both use-cases).
Once I noticed the "computed: True" in the environment resource I figured it was fine to have two variables.
(One is just computed and can never be set by the user).

I'm going very slow with these changes (supporting template less environment) due to all these small nuances.
Some of it is not straightforward.